examples/viewer/src/main.rs

```diff
@@ -94,146 +107,148 @@ fn main() {
+            // Get the winit window cached in a previous Suspended event or else create a new window
+            let window = cached_window
+                .take()
+                .unwrap_or_else(|| create_winit_window(event_loop));
 
+            // Create a vello Surface
+            let size = window.inner_size();
+            let surface_future = render_cx.create_surface(
+                window.clone(),
+                size.width,
+                size.height,
+                wgpu::PresentMode::AutoNoVsync,
+            );
             let surface = pollster::block_on(surface_future).expect("Error creating surface");
-            render_state = {
-                let render_state = RenderState { window, surface };
-                renderer = Some(
-                    Renderer::new(
-                        &render_cx.devices[render_state.surface.dev_id].device,
-                        &RendererOptions {
-                            surface_format: Some(render_state.surface.format),
-                            timestamp_period: render_cx.devices[render_state.surface.dev_id]
-                                .queue
-                                .get_timestamp_period(),
-                            use_cpu: false,
-                        },
-                    )
-                    .expect("Could create renderer"),
-                );
-                Some(render_state)
-            };
-            *control_flow = ControlFlow::Poll;
+
+            // Create a vello Renderer for the surface (using its device id)
+            renderers.resize_with(render_cx.devices.len(), || None);
+            renderers[surface.dev_id]
+                .get_or_insert_with(|| create_vello_renderer(&render_cx, &surface));
+
+            // Save the Window and Surface to a state variable
+            render_state = RenderState::Active(ActiveRenderState { window, surface });
+
+            event_loop.set_control_flow(ControlFlow::Poll);
         }
+
         _ => {}
     });
 }
+
+/// Helper function that creates a Winit window and returns it (wrapped in an Arc for sharing between threads)
+fn create_winit_window(event_loop: &winit::event_loop::EventLoopWindowTarget<()>) -> Arc<Window> {
+    Arc::new(
+        WindowBuilder::new()
+            .with_inner_size(LogicalSize::new(1044, 800))
+            .with_resizable(true)
+            .with_title("Vello Shapes")
+            .build(event_loop)
+            .unwrap(),
+    )
+}
+
+/// Helper function that creates a vello Renderer for a given RenderContext and Surface
+fn create_vello_renderer(render_cx: &RenderContext, surface: &RenderSurface) -> Renderer {
+    Renderer::new(
+        &render_cx.devices[surface.dev_id].device,
+        RendererOptions {
+            surface_format: Some(surface.format),
+            use_cpu: false,
+            antialiasing_support: vello::AaSupport::all(),
+            num_init_threads: NonZeroUsize::new(1),
```

Comment:
```suggestion
            num_init_threads: None,
```

If you set it to None, Vello will use a heuristic which will use many but not all threads. This would be the sensible default.